deepchecks/base/check_context.py

```diff
@@ -0,0 +1,181 @@
+                 ):
+        # Validations
+        if train and isinstance(train, pd.DataFrame):
+            train = Dataset(train, cat_features=[])
+        if test and isinstance(test, pd.DataFrame):
+            test = Dataset(test, cat_features=[])
+        # If both datasets, validate they fit each other
+        if train and test:
+            if not Dataset.datasets_share_label(train, test):
+                raise DatasetValidationError('train and test requires to have and to share the same label')
+            if not Dataset.datasets_share_features(train, test):
+                raise DatasetValidationError('train and test requires to share the same features columns')
+            if not Dataset.datasets_share_categorical_features(train, test):
+                raise DatasetValidationError(
+                    'train and test datasets should share '
+                    'the same categorical features. Possible reason is that some columns were'
+                    'inferred incorrectly as categorical features. To fix this, manually edit the '
+                    'categorical features using Dataset(cat_features=<list_of_features>'
+                )
+            if not Dataset.datasets_share_index(train, test):
+                raise DatasetValidationError('train and test requires to share the same index column')
+
+        if train and model:
+            validate_model(train, model)
+
+        self._train = train
+        self._test = test
+        self._model = model
+        self._feature_importance_mode = feature_importance_mode
+        self._features_importance = None
+        self._task_type = None
+        self._user_scorers = scorers
+        self._user_non_avg_scorers = non_avg_scorers
+
+    # Properties
+    # Validations note: We know train & test fit each other so all validations can be run only on train
+
+    @property
+    def train(self):
+        if self._train is None:
+            raise DatasetValidationError('Check is irrelevant for Datasets without train dataset')
+        return self._train
+
+    @property
+    def test(self):
+        if self._test is None:
+            raise DatasetValidationError('Check is irrelevant for Datasets without test dataset')
+        return self._test
+
+    @property
+    def model(self):
+        if self._model is None:
+            raise DatasetValidationError('Check is irrelevant for Datasets without model')
+
+    @property
+    def label_name(self):
+        if self.train.label_name is None:
+            raise DatasetValidationError('Check is irrelevant for Datasets without label')
+        return self.train.label_name
+
+    @property
+    def features(self):
+        if not self.train.features_columns:
+            raise DatasetValidationError('Check is irrelevant for Datasets without features')
+        return self.train.features
+
+    @property
+    def date_name(self):
+        if self.train.datetime_name is None:
```

Comment:
not correct (same for index) - datetime can be in dataframe index. Current way of checking if datetime is not set is by checking if self.datetime_col is None.

also, call this datatime